src/impl_raw_views.rs

```diff
@@ -14,7 +16,7 @@ where
     pub(crate) unsafe fn new_(ptr: *const A, dim: D, strides: D) -> Self {
         RawArrayView {
             data: RawViewRepr::new(),
-            ptr: ptr as *mut A,
+            ptr: NonNull::new(ptr as *mut A).unwrap(),
```

Comment:
This one unwrap I'd seriously consider to make unchecked for release builds (so it would be guarded by a debug assertion otherwise). This path ought to be performance sensitive - also a code bloat issue.

I think there would be a few similar cases other places in the PR.